src/PointwiseFunctions/SpecialRelativity/LorentzBoostMatrix.hpp

```diff
@@ -0,0 +1,54 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+
+/// \cond
+namespace gsl {
+template <typename T>
+class not_null;
+}  // namespace gsl
+
+/// \endcond
+
+/// \ingroup SpecialRelativityGroup
+/// Holds functions related to special relativity.
+namespace sr {
+// @{
+/*!
+ * \ingroup SpecialRelativityGroup
+ * \brief Computes the matrix for a Lorentz boost from a single
+ * velocity vector (i.e., not a velocity field).
+ *
+ * \details Given a spatial velocity vector \f$v^i\f$ (with \f$c=1\f$),
+ * compute the matrix \f$\Lambda^{a}{}_{\bar{a}}\f$ for a Lorentz boost with
+ * that velocity [e.g. Eq. (2.38) of \cite ThorneBlandford2017]:
+ *
+ * \f{align}{
+ * \Lambda^t{}_{\bar{t}} &= \gamma, \\
+ * \Lambda^t{}_{\bar{i}} = \Lambda^i{}_{\bar{t}} &= \gamma v^i, \\
+ * \Lambda^i{}_{\bar{j}} = \Lambda^j{}_{\bar{i}} &= [(\gamma - 1)/v^2] v^i v^j
+ *                                              + \delta^{ij}.
+ * \f}
+ *
+ * Here \f$v = \sqrt{\delta_{ij} v^i v^j}\f$, \f$\gamma = 1/\sqrt{1-v^2}\f$,
+ * and \f$\delta^{ij}\f$ is the Kronecker delta.
+ *
+ * Note that while the Lorentz boost matrix is symmetric, the returned
+ * boost matrix is of type `tnsr::Ab`, because `Tensor` does not support
+ * symmetric tensors unless both indices have the same valence.
+ */
+template <size_t SpatialDim, typename Frame>
+tnsr::Ab<double, SpatialDim, Frame> lorentz_boost_matrix(
```

Comment:
There are only multiple frames if you think of the boost as a passive operation, but you can equivalently think of it as active (changing the velocity of something, but only having one frame). I figure since only the former involves multiple frames, it would be better to leave it to the user to specify the frame of the objects being acted on. Also, because a `Tensor` can only have one frame, it's not clear to me what frame I would give to boost_matrix if I took the passive view.

I used `Tensor` because it was easiest to get going and because it supports symmetry and compile-time indexing, and because someone might someday want to generalize to template on datatype (but not in this PR). If there's a consensus to use `Matrix` or `DenseMatrix`, I'm happy to make the change

Reply:
A `Tensor` can specify a different frame on each index (see e.g. Jacobian/InvJacobian).